nav2_theta_star_planner/src/theta_star_planner.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020 Anshumaan Singh
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include <memory>
+#include <string>
+#include "nav2_theta_star_planner/theta_star_planner.hpp"
+#include "nav2_theta_star_planner/theta_star.hpp"
+
+namespace nav2_theta_star_planner
+{
+void ThetaStarPlanner::configure(
+  const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent,
+  std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros)
+{
+  planner_ = std::make_unique<theta_star::ThetaStar>();
+  auto node_ = parent.lock();
+  logger_ = node_->get_logger();
+  clock_ = node_->get_clock();
+  name_ = name;
+  tf_ = tf;
+  planner_->costmap_ = costmap_ros->getCostmap();
+  global_frame_ = costmap_ros->getGlobalFrameID();
+
+  nav2_util::declare_parameter_if_not_declared(
+	  node_, name_ + ".how_many_corners", rclcpp::ParameterValue(
+      8));
+
+  node_->get_parameter(name_ + ".how_many_corners", planner_->how_many_corners_);
+
+  if (planner_->how_many_corners_ != 8 && planner_->how_many_corners_ != 4) {
+    planner_->how_many_corners_ = 8;
+    RCLCPP_WARN(logger_,
+				"Your value for - .how_many_corners  was overridden, and is now set to 8");
+  }
+
+  nav2_util::declare_parameter_if_not_declared(
```

Comment:
Are all of these values reasonable defaults?

Reply:
Yes, I have set them to the values for which I have got paths I had shown to you earlier